pkg/build/dockerfile-generator.go

```diff
@@ -45,9 +47,20 @@ func (g *DockerfileGenerator) GenerateDockerFile(ctx context.Context) error {
 	ctx, span := tracing.StartSpan(ctx)
 	defer span.EndSpan()
 
-	lines, err := g.buildDockerfile(ctx)
-	if err != nil {
-		return span.Error(fmt.Errorf("error generating the Dockerfile: %w", err))
+	var lines []string
+	var err error
+	if g.Config.IsFeatureEnabled(experimental.FlagFullControlDockerfile) {
+		span.Warnf("WARNING: Experimental feature \"%s\" enabled: Dockerfile will be used without changes by Porter",
+			experimental.FullControlDockerfile)
+		lines, err = g.readRawDockerfile(ctx)
+		if err != nil {
+			return span.Error(fmt.Errorf("error reading the Dockerfile: %w", err))
+		}
+	} else {
```

Comment:
Do we need this else? I think we can just delete it and have the same functinality.

Reply:
I'm not sure I follow you.
If we remove the else and keep the call to `g.buildDockerfile(ctx)` the final contents will be "raw dockerfile contents" + "dockerfile contents with templating and porter magic". 
If we remove the else block completely then the experimental feature works but the regular Joes will end up with empty Dockerfiles.

Reply:
I misread and thought the function ended right under it, today I learned how to scroll